src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -379,7 +388,11 @@ HTTPProxy::_handleCreateQuery(shd::CreateQueryRequest&& request) {
             }
 
             auto queryId = _queryID++;
-            _txns.at(req.timestamp).queries[queryId] = std::move(result.query);
+            try {
```

Comment:
This works, but the rest of our codebase uses the iterator pattern instead of exception-based checks for existence. Please use iterator check instead:
`
if (auto it = _txns.find(req.timestamp); it == _txns.end()) {
   // send error
}
else {
   // use *it
}
`